File: PRACTICA 5/EcuacionCuadratica-Estructurada.py
import math

class EcuacionCuadratica:
    def __init__(self, a, b, c):
        self.a = a
        self.b = b
        self.c = c

    def get_discriminante(self):
        return (self.b ** 2) - (4 * self.a * self.c)

    def get_raiz1(self):
        d = self.get_discriminante()
        return (-self.b + math.sqrt(d)) / (2 * self.a) if d >= 0 else None

    def get_raiz2(self):
        d = self.get_discriminante()
        return (-self.b - math.sqrt(d)) / (2 * self.a) if d >= 0 else None

    def __str__(self):
        d = self.get_discriminante()
        if d > 0:
            return f"La ecuación tiene dos raíces: {self.get_raiz1()} y {self.get_raiz2()}"
        elif d == 0:
            return f"La ecuación tiene una raíz: {self.get_raiz1()}"
        else:
            return "La ecuación no tiene raíces reales"

def main():
    try:
        a = float(input("Ingrese a: "))
        b = float(input("Ingrese b: "))
        c = float(input("Ingrese c: "))
        ecuacion = EcuacionCuadratica(a, b, c)
        print(ecuacion)
    except ValueError:
        print("Entrada inválida. Debe ingresar números.")

if __name__ == "__main__":
    main()

File: PRACTICA 5/JAVA/EcuacionCuadratica-estructurada.java
import java.util.Scanner;

class EcuacionCuadratica {
    private double a, b, c;
    public EcuacionCuadratica(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    public double getDiscriminante() {
        return (b * b) - (4 * a * c);
    }

    public Double getRaiz1() {
        double discriminante = getDiscriminante();
        return discriminante >= 0 ? (-b + Math.sqrt(discriminante)) / (2 * a) : null;
    }

    public Double getRaiz2() {
        double discriminante = getDiscriminante();
        return discriminante >= 0 ? (-b - Math.sqrt(discriminante)) / (2 * a) : null;
    }

    @Override
    public String toString() {
        double discriminante = getDiscriminante();
        if (discriminante > 0) {
            return "La ecuación tiene dos raíces: " + getRaiz1() + " y " + getRaiz2();
        } else if (discriminante == 0) {
            return "La ecuación tiene una raíz: " + getRaiz1();
        } else {
            return "La ecuación no tiene raíces reales";
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese a, b, c: ");
        double a = sc.nextDouble();
        double b = sc.nextDouble();
        double c = sc.nextDouble();
        sc.close();

        EcuacionCuadratica ecuacion = new EcuacionCuadratica(a, b, c);
        System.out.println(ecuacion);
    }
}
